src/cockpit/beiboot.py

```diff
@@ -212,60 +245,68 @@ def __init__(self, router: Router, destination: str, args: argparse.Namespace):
             if user:  # this can be empty, i.e. auth is just ":"
                 logger.debug("got username %s and password from Basic auth", user)
-                ssh_opts = ['-l', user]
+                username_opts = ['-l', user]
 
-        beiboot_helper = BridgeBeibootHelper(self)
+        # We want to run a python interpreter somewhere...
+        cmd, env = python_interpreter('cockpit-bridge')
 
-        agent = ferny.InteractionAgent([AuthorizeResponder(self.router, basic_password), beiboot_helper])
+        # outside of the flatpak we expect cockpit-ws and thus an installed helper
+        askpass = patch_libexecdir('${libexecdir}/cockpit-askpass')
+        assert isinstance(askpass, str)
+        ssh_askpass = Path(askpass)
+        if not ssh_askpass.exists():
+            logger.error("Could not find cockpit-askpass helper at %r", askpass)
 
-        # We want to run a python interpreter somewhere...
-        cmd: Sequence[str] = ('python3', '-ic', '# cockpit-bridge')
-        env: Sequence[str] = ()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            user_known_hosts_file = Path(tmpdir) / 'user-known-hosts'
+            user_known_hosts_file.write_text(user_known_hosts)
 
-        in_flatpak = os.path.exists('/.flatpak-info')
+            cmd, env = via_ssh(cmd, self.destination, ssh_askpass, *username_opts,
+                               '-o', f'UserKnownHostsfile={user_known_hosts_file!s}')
 
-        # Remote host?  Wrap command with SSH
-        if self.destination != 'localhost':
-            if in_flatpak:
-                # we run ssh and thus the helper on the host, always use the xdg-cache helper
-                ssh_askpass = ensure_ferny_askpass()
-            else:
-                # outside of the flatpak we expect cockpit-ws and thus an installed helper
-                askpass = patch_libexecdir('${libexecdir}/cockpit-askpass')
-                assert isinstance(askpass, str)
-                ssh_askpass = Path(askpass)
-                if not ssh_askpass.exists():
-                    logger.error("Could not find cockpit-askpass helper at %r", askpass)
-
-            env = (
-                f'SSH_ASKPASS={ssh_askpass!s}',
-                'DISPLAY=x',
-                'SSH_ASKPASS_REQUIRE=force',
-            )
-            host, _, port = self.destination.rpartition(':')
-            # catch cases like `host:123` but not cases like `[2001:abcd::1]
-            if port.isdigit():
-                ssh_opts += ['-p', port, host]
-            else:
-                ssh_opts += [self.destination]
+            await self.boot(cmd, env, basic_password)
 
-            cmd = ('ssh', *ssh_opts, shlex.join(cmd))
+            self.router.write_control(
+                command='authorize', challenge='x-login-data', cookie='-', login_data={
+                    'user-known-hosts': user_known_hosts_file.read_text()
+                }
+            )
```

Comment:
This doesn't fit here in "split flatpak and bastion code paths", it should be part of the next commit ("support for temporary UserKnownHostsFile")

Reply:
rebase fail :)

Reply:
Moved.